text/0000-partial_types.md

```diff
@@ -0,0 +1,642 @@
+
+The rest of omitted `%field-accessN` field-accesses means `%same` (same as previous `%field-accessN-1`)
+
+| `%field-accessN-1` | `%same`   | `%opposite` |
+|--------------------|-----------|-------------|
+| `%permit`          | `%permit` | `%deny`     |
+| `%deny`            | `%deny`   | `%permit`   |
+| `%miss` (E)        | `%miss`   | `%permit`   |
+
+Examples:
+```rust
+struct Point {x: f64, y: f64, was_x: f64, was_y: f64}
+
+let pfull = Point {was_x: 4.0, was_y: 5.0};
+	// pfull : Point.%{was_x, was_y, %miss x, y}
+```
+
+Type access `.%full` is a shortcut for `.%{*}`.
+
+It is forbidden to have `.%{%deny *}` values and `.%{%deny *}` references (but who knows, maybe they are suitable uninhabited types, phantom types or proxy types).
+
+## Access-filter and Mutable Filter
+
+Rust consumer use same names for action and for type clarifications, so we follow this style.
+
+Access-filter is an action, which is looks like detailed access and it is written left to variable at consuming (moving, borrowing, referencing, initializing, returning, pick fields).
+
+| ↓filter / →var access | `%permit` | `%deny`   | `%miss` (E) | `%ignore` (D) |
+|-----------------------|-----------|-----------|-------------|---------------|
+| `%permit`             | `%permit` | !ERROR    | !ERROR      | !ERROR        |
+| `%deny`               | `%deny`   | `%deny`   | `%deny`     | `%deny`       |
+| `%miss` (E)           | !ERROR    | !ERROR    | `%miss`     | !ERROR        |
+| `%ignore` (F)         | !ERROR    | !ERROR    | !ERROR      | `%ignore`     |
+
+It is allowed to write explicitly specific access-filter with same rules as detailed access.
+
+Default filter on consumption if filter is omitted for:
+ - `var  ~  var.%max`
+ - `& var  ~  & var.%max`
+ - `&mut var  ~  &mut var.%max`
+ 
+(A + D)
+ - `return var  ~  return var.%exact`
+ - `return & var  ~  return & var.%exact`
+ - `return &mut var  ~ return &mut var.%exact`
+ 
+ where filters are based on variable field-access:
+
+| ↓var access  | `.%max`   | `.%exact` (D) |
+|--------------|-----------|---------------|
+| `%permit`    | `%permit` | `%permit`     |
+| `%deny`      | `%deny`   | `%deny`       |
+| `%miss` (E)  | `%miss`   | `%miss`       |
+| `%ignore`    | `%deny`   | `%ignore`     |
+
+## New Picker several fields
+
+Rust has special syntax for "pick one field" (get field-access):
+```rust
+struct Point {x: f64, y: f64, was_x: f64, was_y: f64}
+
+let mut pfull = Point {was_x: 4.0, was_y: 5.0};
+
+let px = pfull.x;
+let rpy = & pfull.y;
+let bpwasx = &mut pfull.was_x;
+```
+
+But it is still impossible to "pick several fields". With partial types it become possible. 
+
+I suggest to add additional picker for variables `var.{fld1, fld2, ..}` (similar to partiality `var.%{fld1, fld2}`):
+```rust
+let pxy = pfull.{x, y};
+//   same as
+let pxy = pfull.%{x, y};
```

Comment:
Why add two syntaxes that do the same thing? Are you trying to make this feature as difficult to learn as possible?

Reply:
Thank you, I'll cut this. First remains:
```rust
let pxy = pfull.{x, y};
```

Reply:
Do we need this at all? Can't we just solve this with intra-procedural analysis, just as is done for lifetimes and partial borrowing today? You don't need to write `let r = &'a x;`, the compiler just figures out what `'a` should be based on how it is used. Only types really require this annotation (like lifetimes but not partial borrowing today).

Reply:
@digama0 good question!
If we use such variables as argument to to the function or method, then yes, we can solve, which partiality is required.
But in rest of cases it is hard to say.
Hopefully, main specialization of partial types - is to be used as arguments!
So, if we add "implicit partiality" for arguments - we cover most useful cases

Reply:
like I said, the compiler *already does* implicit partiality. The only issue that needs solving here is that there is no syntax for *explicit* partiality - the compiler can understand when you late initialize a field or borrow part of a struct: this is already expressible.

Reply:
@digama0  Do you suggest "use implicit rules for this if this do not use explicit rules" for partiality. Good point!